### PR TITLE
fix issue for set workload revisionHistoryLimit [bsc#1162651]

### PR DESCRIPTION
**Pull request by cclhsu**
fix issue items for set workload revisionHistoryLimit [bsc#1162651]

Signed-off-by: cclhsu <clark.hsu@suse.com>

## Why is this PR needed?

Reduce revisionHistoryLimit to as decribed in https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#revision-history-limit.  Per discussion, [#1162651](https://bugzilla.suse.com/show_bug.cgi?id=1162651), change our revisionHistoryLimit of our workloads from 10 to 3.
 
```
# kubectl -n kube-system get deployments,replicasets,daemonset -o jsonpath='{range .items[*]}{@.kind}{" "}{@.metadata.name}{" "}{@.spec.revisionHistoryLimit}{"\n"}'

Deployment cilium-operator 10
Deployment oidc-dex 10
Deployment oidc-gangway 10
DaemonSet cilium 10
DaemonSet kured 10
```

Fixes [#1162651](https://bugzilla.suse.com/show_bug.cgi?id=1162651)

## What does this PR do?

```
# kubectl -n kube-system get deployments,replicasets,daemonset -o jsonpath='{range .items[*]}{@.kind}{" "}{@.metadata.name}{" "}{@.spec.revisionHistoryLimit}{"\n"}'

Deployment cilium-operator 3
Deployment oidc-dex 3
Deployment oidc-gangway 3
DaemonSet cilium 3
DaemonSet kured 3
```

## Anything else a reviewer needs to know?

Special test cases, manual steps, links to resources or anything else that could be helpful to the reviewer.

## Info for QA

This is info for QA so that they can validate this. This is **mandatory** if this PR fixes a bug.
If this is a new feature, a good description in "What does this PR do" may be enough.

### Related info

Info that can be relevant for QA:
* link to other PRs that should be merged together
* link to packages that should be released together
* upstream issues

### Status **BEFORE** applying the patch

How can we reproduce the issue? How can we see this issue? Please provide the steps and the prove
this issue is not fixed.

### Status **AFTER** applying the patch

How can we validate this issue is fixed? Please provide the steps and the prove this issue is fixed.

## Docs

If docs need to be updated, please add a link to a PR to https://github.com/SUSE/doc-caasp.
At the time of creating the issue, this PR can be work in progress (set its title to [WIP]),
but the documentation needs to be finalized before the PR can be merged. 

# Merge restrictions

(Please do not edit this)

We are in *v4-maintenance phase*, so we will restrict what can be merged to prevent unexpected surprises:

    What can be merged (merge criteria):
        2 approvals:
            1 developer: code is fine
            1 QA: QA is fine
        there is a PR for updating documentation (or a statement that this is not needed)

### CODE REVIEW

**Comment by c3y1huang** [comment]
Think missing manifest version bump. 

And could we add some test-case along side this?

File: internal/pkg/skuba/kubernetes/versions.go [hunk 1]
```diff
@@ -111,10 +111,10 @@ var (
...
 			},
 			AddonsVersion: AddonsVersion{
-				Cilium:  &AddonVersion{"1.5.3", 1},
-				Kured:   &AddonVersion{"1.2.0-rev4", 1},
-				Dex:     &AddonVersion{"2.16.0", 4},
-				Gangway: &AddonVersion{"3.1.0-rev4", 3},
+				Cilium:  &AddonVersion{"1.5.3", 2},
+				Kured:   &AddonVersion{"1.2.0-rev4", 2},
+				Dex:     &AddonVersion{"2.16.0", 5},
+				Gangway: &AddonVersion{"3.1.0-rev4", 4},
```
Comment (jenting) [inline internal/pkg/skuba/kubernetes/versions.go]: IMO, I don't think we need to bump K8s 1.15.2 cluster addons manifest version.

File: internal/pkg/skuba/addons/dex.go [hunk 2]
```diff
@@ -164,6 +164,7 @@ metadata:
   labels:
     app: oidc-dex
 spec:
+  revisionHistoryLimit: 3
```
Comment (jenting) [inline internal/pkg/skuba/addons/dex.go]: Already in https://github.com/SUSE/skuba/blob/c8998e8/internal/pkg/skuba/addons/dex.go#L181

File: internal/pkg/skuba/addons/gangway.go [hunk 3]
```diff
@@ -125,6 +125,7 @@ metadata:
   labels:
     app: oidc-gangway
 spec:
+  revisionHistoryLimit: 3
```
Comment (jenting) [inline internal/pkg/skuba/addons/gangway.go]: Already in https://github.com/SUSE/skuba/blob/c8998e8/internal/pkg/skuba/addons/gangway.go#L139

File: internal/pkg/skuba/kubernetes/versions.go (hunk 1 above)
Comment (c3y1huang) [reply internal/pkg/skuba/kubernetes/versions.go]: I couldn't think of a reason customer installed with K8s 1.15.2 shouldn't also benefit from this fix.

Comment (jenting) [reply internal/pkg/skuba/kubernetes/versions.go]: From the product perspective.

AFAIK, we don't have to take 4.0.x once 4.1.x released. The customer has to upgrade from 4.0.x to 4.1.x first.

On the other hand, I saw kured bump container image version only on `1.16.2`. Intended or missing?

Comment (c3y1huang) [reply internal/pkg/skuba/kubernetes/versions.go]: > From the product perspective.
> 
> AFAIK, we don't have to take 4.0.x once 4.1.x released. The customer has to upgrade from 4.0.x to 4.1.x first.

Is upgrade to K8s 1.16.2 is enforced if customer have cluster with 1.15.2? Or could customer upgrade addon only?

> 
> On the other hand, I saw kured bump container image version only on `1.16.2`. Intended or missing?

Comment (jenting) [reply internal/pkg/skuba/kubernetes/versions.go]: I think the customer could upgrade addon only~

Comment (innobead) [reply internal/pkg/skuba/kubernetes/versions.go]: Yep, it looks like if would like to benefit for the current cluster w/o cluster upgrade, we should bump addon version for what cluster version we would like to support.

So it should be fine for this change.

https://github.com/SUSE/skuba/blob/c5895c7bfc9fd5fdaa10e6997bb52364b7f69e43/pkg/skuba/actions/addon/upgrade/plan.go#L51

**Comment by c3y1huang** [comment]
There is still a on going [discussion](https://bugzilla.suse.com/show_bug.cgi?id=1162651#c5) about what revision should be set for cillium* .

**Comment by jordimassaguerpla** [comment]
This is blocking packaging for 4.2.0. Any ETA on this? Does it really need to be in 4.2.0? If so, why?

**Comment by c3y1huang** [comment]
@cclhsu since this is a bug fix. Could you add test case along side this?

**Comment by innobead** [comment]
Good reminder! Chinya :)

**Pull request merged**